backends-velox/src/test/scala/org/apache/gluten/execution/MiscOperatorSuite.scala

```diff
@@ -2092,4 +2092,38 @@ class MiscOperatorSuite extends VeloxWholeStageTransformerSuite with AdaptiveSpa
         }
       })
   }
+
+  test("Test factorial function correctness") {
```

Comment:
Better to move this new test to ScalarFunctionsValidateSuite.scala. Please also rename the test to "factorial function" for brevity. No need to redundantly add "Test" in the name.